htdocs/document.php

```diff
@@ -207,19 +207,18 @@ function llxFooter()
 
 
 // Find the subdirectory name as the reference
-$refname = basename(dirname($original_file)."/");
+//$refname = basename(dirname($original_file)."/");
 
 // Security check
 if (empty($modulepart)) {
 	accessforbidden('Bad value for parameter modulepart');
 }
 
 // Check security and set return info with full path of file
-$check_access = dol_check_secure_access_document($modulepart, $original_file, $entity, $user, $refname);
+$check_access = dol_check_secure_access_document($modulepart, $original_file, $entity, $user, '');
```

Comment:
This introduce a security hole by allowing to download file without checking on object itself (just on permission on module) so no check on entity become possible.